broadcast.go

```diff
@@ -96,6 +96,18 @@ func (b *Broadcaster) Close() error {
 	}
 }
 
+// Done returns a channel that will always proceed once the broadcaster is
+// closed.
+func (b *Broadcaster) Done() <-chan struct{} {
+	ch := make(chan struct{})
+	select {
+	case <-b.closed:
+		close(ch)
+	default:
+	}
+	return ch
```

Comment:
Can you not return `b.closed` directly? This won't update if it is closed in the interim.